metaseq/tasks/base_task.py

```diff
@@ -485,7 +485,7 @@ def reduce_metrics(self, logging_outputs, criterion):
 
         # This should only be activated if we are doing data pruning for language modeling evaluation
         if self.args._name == "streaming_language_modeling" and ('compute_data_pruning_metrics' in self.args and self.args['compute_data_pruning_metrics']):
-            criterion.__class__.reduce_metrics(logging_outputs, self.data_pruning_metrics, self.data_pruning_savedir)
+            criterion.__class__.reduce_metrics(logging_outputs, self.data_pruning_metrics, self.data_pruning_savedir, len(self.datasets['train'].dataset))
```

Comment:
Is this something we could set when we initialize the criterion? Just feels a bit odd to change code in places like this.

Reply:
I also had this thought; the reason I didn't do it this way is because this is a `staticmethod` (so I assumed it can't rely on attributes of the criterion object, and we should pass arguments directly to the function - if this is wrong, I can change it).